etl/src/main/java/org/pih/hivmigration/etl/sql/ReturnVisitDateMigrator.groovy

```diff
@@ -0,0 +1,65 @@
+package org.pih.hivmigration.etl.sql
+
+class ReturnVisitDateMigrator extends SqlMigrator{
+
+    @Override
+    def void migrate() {
+
+        executeMysql("Create staging table for migrating next scheduled visit date", '''
+            create table hivmigration_next_visit_date (                            
+              source_encounter_id int,
+              source_patient_id int,
+              next_exam_date date            
+            );
+        ''')
+
+        loadFromOracleToMySql('''
+            insert into hivmigration_next_visit_date (
+              source_encounter_id,
+              source_patient_id,
+              next_exam_date 
+            )
+            values(?,?,?) 
+            ''', '''
+                select x.encounter_id as source_encounter_id,
+                        e.patient_id as source_patient_id,
+                        to_char(x.next_exam_date, 'yyyy-mm-dd') as next_exam_date
+                from hiv_exam_extra x,hiv_encounters e 
+                where x.next_exam_date is not null and x.encounter_id = e.encounter_id
+            ''')
+
+        executeMysql("Create tmp_obs_table", ''' 
+            CALL create_tmp_obs_table();
+            ''')
+
+        setAutoIncrement("tmp_obs", "(select max(obs_id)+1 from obs)")
+
+        executeMysql("Load next visit date as observations", ''' 
+                      
+            SET @next_visit_date_concept_uuid = '3ce94df0-26fe-102b-80cb-0017a47871b2';            
+            
+            -- Do not overwrite existing Next Dispense Date obs that were inserted while migrating meds
+            INSERT INTO tmp_obs (value_datetime, source_patient_id, source_encounter_id, concept_uuid)
+            SELECT next_exam_date, source_patient_id, source_encounter_id, @next_visit_date_concept_uuid
+            FROM hivmigration_next_visit_date
+            WHERE next_exam_date is not null and source_encounter_id not in (select e.source_encounter_id from hivmigration_encounters e, obs o, concept c where e.encounter_id = o.encounter_id and o.concept_id=c.concept_id and c.uuid=@next_visit_date_concept_uuid);
```

Comment:
NO.  We don't want to add this additional where clause and possibly exclude data.  These are different data points and we need both.  If there is a risk of overlap, then we need to deal with this by altering the way we are modeling the data.